participants/olivercoad/simple_functions.py

```diff
@@ -0,0 +1,26 @@
+def fibonacci(max):
+    values = [0, 1]
+    while values[-2] + values[-1] < max:
+        values.append(values[-2] + values[-1])
+    return values
+
+
+def factorial(value):
+    if value == 0:
+        return 1
+    else:
+        return value * factorial(value - 1)
+
+# from https://en.wikipedia.org/wiki/Primality_test#Python_code
+def is_prime(n: int) -> bool:
+    """Primality test using 6k+-1 optimization."""
+    if n <= 3:
+        return n > 1
+    if n % 2 == 0 or n % 3 == 0:
+        return False
+    i = 5
+    while i ** 2 <= n:
+        if n % i == 0 or n % (i + 2) == 0:
+            return False
+        i += 6
+    return True
```

Comment:
Looks good, interesting test for primality I haven't seen before.

Reply:
Yes it's a neat optimization. There was a Numberphile video a couple years ago about it.